pkg/raft/raft.go

```diff
@@ -1066,12 +1066,18 @@ func (r *raft) reset(term uint64) {
 
 	r.abortLeaderTransfer()
 
-	r.electionTracker.ResetVotes()
 	r.trk.Visit(func(id pb.PeerID, pr *tracker.Progress) {
+		guessMatch := r.raftLog.lastIndex()
+		hintEntry, ok := r.electionTracker.GetHintEntry(id)
+		conflictIndex, t := r.raftLog.findConflictByTerm(hintEntry.Index, hintEntry.Term)
+		if ok && t != 0 {
+			guessMatch = conflictIndex
+		}
+
 		*pr = tracker.Progress{
```

Comment:
This still initializes all flows to `StateProbe`, but the point of all this was also to initialize to `StateReplicate` in best case.

The two modifications can be done in separate commits though:
- in this commit we improve the initial guess (by setting `Next`)
- in another commit we can also enter `StateReplicate` for those flows for which we have a match